bundles/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/bus/KNXBinding.java

```diff
@@ -108,7 +108,7 @@ public void removeKNXTypeMapper(KNXTypeMapper typeMapper) {
     @Override
     protected void internalReceiveCommand(String itemName, Command command) {
         logger.trace("Received command (item='{}', command='{}')", itemName, command.toString());
-        if (!isEcho(itemName, command)) {
+        if (!KNXConnection.getCompatibilityOH2() && !isEcho(itemName, command)) {
```

Comment:
@watou, when many other bindings do it right, then its time to find a way for KNX1 too! ;-)

Lets have a closer look at autoupdate! Seams there could be the fault.
The different behavior between and OH1 and OH2 compat1x is observed with default configured items, autoupdate was not configured explicit.

Found autoupdate in [KNXGenericBindingProvider.autoUpdate](https://github.com/openhab/openhab/blob/master/bundles/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/config/KNXGenericBindingProvider.java#L347)
Why shouldn't this work in OH2? Can you see any faults or better constructs to use here?

I do not understand the whole *autoupdate* mechanism comes together with internalReceiveCommand/internalReceiveUpdate, yet? Will have a closer look...


Yes, you are absolute right, *ignoreEventList* can grow unstopped in this way, it has to be flipped:
`if (!isEcho(itemName, command) && !KNXConnection.getCompatibilityOH2()) {
`